src/scenic/syntax/veneer.py

```diff
@@ -111,1000 +132,1341 @@
-			if isinstance(Y, OrientedPoint):
-				raise RuntimeParseError('"X relative to Y" with X, Y both oriented points')
-			Y = toVector(Y, '"X relative to Y" with X an oriented point but Y not a vector')
-			return X.relativize(Y)
-		elif isinstance(Y, OrientedPoint):
-			X = toVector(X, '"X relative to Y" with Y an oriented point but X not a vector')
-			return Y.relativize(X)
-		else:
-			X = toTypes(X, (Vector, builtins.float),
-			            '"X relative to Y" with X neither a vector nor scalar')
-			Y = toTypes(Y, (Vector, builtins.float),
-			            '"X relative to Y" with Y neither a vector nor scalar')
-			return evaluateRequiringEqualTypes(lambda: X + Y, X, Y,
-											   '"X relative to Y" with vector and scalar')
+    """The :grammar:`<vector field> at <vector>` operator."""
+    if not isA(X, VectorField):
+        raise TypeError('"X at Y" with X not a vector field')
+    Y = toVector(Y, '"X at Y" with Y not a vector')
+    return X[Y]
+
+def RelativeTo(X, Y) -> typing.Union[Vector, builtins.float]:
+    """The :scenic:`{X} relative to {Y}` polymorphic operator.
+
+    Allowed forms::
+
+        <value> relative to <value> # with at least one a field, the other a field or heading
+        <vector> relative to <oriented point> # and vice versa
+        <vector> relative to <vector>
+        <heading> relative to <heading>
+    """
+    if isA(X, VectorField) or isA(Y, VectorField):
+        xf, yf = isA(X, VectorField), isA(Y, VectorField)
+        if xf and yf and X.valueType != Y.valueType:
+            raise TypeError('"X relative to Y" with X, Y fields of different types')
+        fieldType = X.valueType if xf else Y.valueType
+        error = '"X relative to Y" with field and value of different types'
+        def helper(context):
+            pos = context.position.toVector()
+            xp = X[pos] if xf else toType(X, fieldType, error)
+            yp = Y[pos] if yf else toType(Y, fieldType, error)
+            return yp + xp
+        return DelayedArgument({'position'}, helper)
+
+    elif isA(X, OrientedPoint):
+        if isA(Y, OrientedPoint):
+            raise TypeError('"X relative to Y" with X, Y both oriented points')
+        Y = toVector(Y, '"X relative to Y" with X an oriented point but Y not a vector')
+        return X.relativize(Y)
+
+    elif isA(Y, OrientedPoint):
+        X = toVector(X, '"X relative to Y" with Y an oriented point but X not a vector')
+        return Y.relativize(X)
+
+    else:
+        @distributionFunction
+        def relativeToCoerceHelper(X, Y):
+            if canCoerce(X, Vector):
+                xf = toVector(X)
+                yf = toVector(Y, '"X relative to Y" with X a vector but Y not a vector')
+
+                return xf + yf
+
+            elif canCoerce(X, Heading):
+                xf = toHeading(X)
+                yf = toHeading(Y, '"X relative to Y" with X a heading but Y not a heading')
+
+                return xf + yf
```

Comment:
Hmm, should we support `Orientation` here? The vector field case above allows the vector field to return an `Orientation`. It's odd to allow `30 deg relative to roadDirection` but not `30 deg relative to (roadDirection at ...)`. The documentation only says `relative to` accepts headings, but I think orientations make sense too (interpreting the first orientation as being with respect to the coordinate system defined by the second orientation).